src/pyrokinetics/diagnostics/synthetic_highk_dbs.py

```diff
@@ -0,0 +1,670 @@
+        R = m.R
+        q = m.q
+
+        # more fancy terms
+        gtheta_cross_gr2 = (R / Jr) ** 2
+        gphi_cross_gr2 = grr / (R**2)
+
+        # gradients of alpha
+        galpha_dot_gr = dalpha_dr * grr + dalpha_dtheta * grtheta
+        galpha2 = (
+            dalpha_dr**2 * grr
+            + 2 * dalpha_dr * dalpha_dtheta * grtheta
+            + dalpha_dtheta**2 * gthetatheta
+            + 1 / R**2
+        )  # |grad(alpha)|**2
+        galpha_cross_gr_2 = (
+            dalpha_dtheta**2 * gtheta_cross_gr2 + gphi_cross_gr2
+        )  # |grad(alpha) x grad(r)|**2
+        bcross_gradr_dot_gradalpha = galpha_dot_gr**2 / (
+            np.sqrt(grr) * np.sqrt(galpha_cross_gr_2)
+        ) - np.sqrt(grr) * galpha2 / np.sqrt(galpha_cross_gr_2)
+
+        # interpolate to thetaloc
+        grr0 = np.interp(thetaloc, theta_grid, grr)
+        grtheta0 = np.interp(thetaloc, theta_grid, grtheta)
+        gthetatheta0 = np.interp(thetaloc, theta_grid, gthetatheta)
+        gzz0 = np.interp(thetaloc, theta_grid, gzz)
+        dalpha_dr0 = np.interp(thetaloc, theta_grid, dalpha_dr)
+        dalpha_dtheta0 = np.interp(thetaloc, theta_grid, dalpha_dtheta)
+        Jr0 = np.interp(thetaloc, theta_grid, Jr)
+        dRdr0 = np.interp(thetaloc, theta_grid, dRdr)
+        dRdtheta0 = np.interp(thetaloc, theta_grid, dRdtheta)
+        dZdr0 = np.interp(thetaloc, theta_grid, dZdr)
+        dZdtheta0 = np.interp(thetaloc, theta_grid, dZdtheta)
+        R0 = np.interp(thetaloc, theta_grid, R)
+        gtheta_cross_gr20 = np.interp(thetaloc, theta_grid, gtheta_cross_gr2)
+        gphi_cross_gr20 = np.interp(thetaloc, theta_grid, gphi_cross_gr2)
+
+        galpha_dot_gr0 = np.interp(thetaloc, theta_grid, galpha_dot_gr)
+        galpha20 = np.interp(thetaloc, theta_grid, galpha2)
+        galpha_cross_gr_20 = np.interp(thetaloc, theta_grid, galpha_cross_gr_2)
+        bcross_gradr_dot_gradalpha0 = np.interp(
+            thetaloc, theta_grid, bcross_gradr_dot_gradalpha
+        )
+
+        # map (kn0, kb0) to (kx0, ky0): [Ruiz PPCF 2022, notes on local mapping in k]. (kn0, kb0) normalized by rhos_loc:
+        self.ky0 = -(q / rho) * kb0 / bcross_gradr_dot_gradalpha0 * rhos_sim / rhos_loc
+        self.kx0 = (
+            kn0 / np.sqrt(grr0) * rhos_sim / rhos_loc
+            + self.ky0 * (rho / q) * galpha_dot_gr0 / grr0
+        )
+
+        # tests
+        R00 = (
+            geo.Rmaj
+        )  # gk_input_data["theta_grid_parameters"]["rmaj"]   # get actual rmaj0 from input file sim
+        eps = rho / R00
+
+        # compute k-resolutions (approx at the OMP): cf. Ruiz Ruiz PPCF 2020, eq. (13)
+        if syn_filter == "gauss":
+            self.dkx0 = 2 / (wR * np.sqrt(grr0)) * rhos_sim / 1e2
+            self.dky0 = 2 * kappa * q / (wZ * dalpha_dtheta0) * rhos_sim / 1e2
+            # elif filter == 'beam':
+            # we need further work on this
+
+    def get_syn_fspec(self, ik, t1, t2, savedir, if_save):
```

Comment:
You can probably just have this run on all the `ky` at once. I think utilising the built in xarray methods would be a good idea too